### PR TITLE
Improve documentation for highContrast, invertColors, and boldText

### PR DESCRIPTION
**Pull request by mvolpato**
Improve the documentation for `highContrast`, `invertColors`, and `boldText`.

While playing with them I was confused by the fact that changing `boldText` had an effect on the app, while changing the other two did not.

Does not solve #78323, but it is a suggested change in that issue.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 1]
```diff
@@ -314,6 +314,18 @@ class MediaQueryData {
 
   /// Whether the device is inverting the colors of the platform.
   ///
+  /// Setting it, for instace by using:
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/media_query.dart:317]: Nit: 
```suggestion
  /// Setting it, for instance by using: 
```

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 2]
```diff
@@ -325,6 +337,18 @@ class MediaQueryData {
   /// Whether the user requested a high contrast between foreground and background
   /// content on iOS, via Settings -> Accessibility -> Increase Contrast.
   ///
+  /// Setting it, for instace by using:
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/media_query.dart:340]: Same here. Rather than duplicating these, you can create a `@template` and re-use it. :)

**Review by Piinks — COMMENTED** [review]
It looks like the analyzer is failing here.

File: packages/flutter/lib/src/widgets/media_query.dart (hunk 2 above)
Comment (mvolpato) [reply packages/flutter/lib/src/widgets/media_query.dart]: Thanks for the comments. I tried `@template` but I could not get it to work as I needed, the code is not the same for the three examples.

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 3]
```diff
@@ -314,6 +314,18 @@ class MediaQueryData {
 
   /// Whether the device is inverting the colors of the platform.
   ///
+  /// Setting it, for instance by using:
+  ///
+  /// ```dart
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/media_query.dart:319]: Can you put these in `{@tool snippet}...`
That way the code will be analyzed and if it becomes out of date we'll know to update it. :)

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 4]
```diff
@@ -314,6 +314,20 @@ class MediaQueryData {
 
   /// Whether the device is inverting the colors of the platform.
   ///
+  /// {@tool snippet}
+  /// Setting it, for instance by using:
+  ///
+  /// ```dart
+  /// MediaQuery(
+  ///   data: MediaQuery.of(context).copyWith(
+  ///     invertColors: false),
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/media_query.dart:323]: nit: just leave this on one line

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 5]
```diff
@@ -314,6 +314,20 @@ class MediaQueryData {
...
+  ///   child: child
+  /// ),
+  /// ```
+  ///
+  /// will have no effect on `child`, and the platform requested value will be used.
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/media_query.dart:328]: all text needs to go above the code in the snippet.

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 6]
```diff
@@ -314,6 +314,20 @@ class MediaQueryData {
 
   /// Whether the device is inverting the colors of the platform.
   ///
+  /// {@tool snippet}
+  /// Setting it, for instance by using:
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/media_query.dart:318]: This snippet is leaving me with more questions.... what are you trying to document with this?

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 7]
```diff
@@ -341,6 +369,20 @@ class MediaQueryData {
   /// Whether the platform is requesting that text be drawn with a bold font
   /// weight.
   ///
+  /// {@tool snippet}
+  /// Setting it, for instance by using:
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/media_query.dart:373]: Same comments as above.

**Comment by goderbauer** [comment]
The analyzer also doesn't appear to be happy with this code.

File: packages/flutter/lib/src/widgets/media_query.dart (hunk 6 above)
Comment (mvolpato) [reply packages/flutter/lib/src/widgets/media_query.dart:318]: You are right, it is not clear. In my opinion, given that the documentation for the three properties `invertColors`, `highContrast`, and `boldText` is very similar, I expected them to behave the same.

You can override the platform's value when you set `boldText`.
You cannot override the platform's value when you set `invertColors` or `highContrast`.

I will change the code by use these words. But if it is unclear only to me then let me know, I will close the PR.

**Comment by goderbauer** [comment]
I am not exactly sure what this change is supposed to clarify now. Seems like its mostly rephrasing what was already documented?

**Comment by mvolpato** [comment]
> I am not exactly sure what this change is supposed to clarify now. Seems like its mostly rephrasing what was already documented?

I have a small quiz for you. Start with a freshly created project `flutter create project`.
You are running on a iOS simulator with default settings (light mode, no accessibility features enabled).

Add `darkTheme`, `highContrastTheme`, and `highContrastDarkTheme` to the `MaterialApp`:
```dart
MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData.light(),
      darkTheme: ThemeData.dark(),
      highContrastTheme: ThemeData.dark(),
      highContrastDarkTheme: ThemeData.light(),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
```
Wrap `Scaffold` with a `MediaQuery` with data:

```dart
MediaQuery.of(context).copyWith(
        highContrast: true,
        invertColors: true,
        boldText: true,
        textScaleFactor: 2.0,
      ),
```

Only by reading the documentation:

1. Which theme is used?
2. Are colors inverted?
3. Is the text bold(er)?
4. Is the text bigger?

Personally, by reading the documentation, I answered wrong to 2 of these questions. If everybody else answers correctly, then I guess it's just me 🤷‍♂️

**Comment by goderbauer** [comment]
What I meant: I don't see how the proposed changes in this PR make answering those questions easier. It probably needs more clarification in the docs to allow a reader to answer those questions correctly?

**Comment by mvolpato** [comment]
> What I meant: I don't see how the proposed changes in this PR make answering those questions easier. It probably needs more clarification in the docs to allow a reader to answer those questions correctly?

I get it. Unfortunately this is the best I can think of. Maybe a native speaker can propose a better wording?

**Comment by goderbauer** [comment]
Looks like we have an issue on file for this. I am going to close this until somebody has time to fix it.

**Pull request closed without merging**